docs/user-guides/client-api-config.md

```diff
@@ -0,0 +1,676 @@
+    // Represent the filter target configuration.
+    repeated Target targets = 1;
+  }
+}
+```
+
+#### timestamp
+
+`timestamp` is the timestamp when the vector is updated.
+When `timestamp` is not set, the current time will be used.
+
+## Upsert Service
+
+`Upsert` is updating existing vectors in the `vald-agent` or inserting new vectors into the `vald-agent` if the request vector is not indexed.
+It requires the vector, its ID (specific ID for the vector), and optional configuration.
+
+### Configuration
+
+```rpc
+// Represent upsert configuration.
+message Config {
+  // Check whether or not the same set of vector and ID is already inserted.
+  bool skip_strict_exist_check = 1;
+  // Configuration for filters if your Vald cluster uses filters.
+  Filter.Config filters = 2;
+  // The timestamp when the vector was inserted.
+  int64 timestamp = 3;
+}
+```
+
+<details><summary>Upsert Configuration Sample (Go)</summary><br>
+
+```go
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/vdaas/vald-client-go/v1/payload"
+	"github.com/vdaas/vald-client-go/v1/vald"
+	"google.golang.org/grpc"
+)
+
+func example() {
+	// Create connection
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	target := "localhost:8080"
+	conn, err := grpc.DialContext(ctx, target)
+	if err != nil {
+		panic(err)
+	}
+	// Init Vald client
+	client := vald.NewValdClient(conn)
+
+	// Upsert sample
+	location, err := client.Upsert(ctx, &payload.Upsert_Request{
+		// Vector information (mandatory)
+		Vector: &payload.Object_Vector{
+			// Set the vector
+			Vector: []float32{0, 1, 2},
+			// Set the specific ID for the vector
+			Id:     "sample",
+		},
+		// Upsert configuration (optional)
+		Config: &payload.Upsert_Config{
+			SkipStrictExistCheck: false,
+			Filters: &payload.Filter_Config{
+				Targets: []*payload.Filter_Target{
+					{
+						Host: "vald-ingress-filter",
+						Port: 8081,
+					},
+				},
+			},
+			Timestamp: time.Now().UnixMilli(),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+        ...
+}
+```
+
+</details>
+
+#### skip_strict_exist_check
+
+`skip_strict_exist_check` (default value is `false`) is a flag to skip checking whether the same set of the vector and ID is already inserted or not.
+If it is set as `false`, the checking function will be skipped.<BR>
+
+When `skip_strict_exist_check` is `false`, the following checking steps will run in the upsert process:
+
+1. Check whether the set of (ID and vector) is already inserted or not.
+   The request ID and vector will be inserted if there is no data.
+1. After passing the step.1, check whether the request vector is the same as the indexed vector.
+   If it is the same, the upsert process ends with returning the `ALREADY_EXIST` error.
+
+The upsert process will continue if all of the above steps have been passed.
+
+#### filters
+
+`filters` is the configuration when using filter functions.
+In the `Upsert` section, it is popular to use ingress filtering.
```

Comment:
Is it better to use `common` ? 🤔 

```suggestion
In the `Upsert` section, it is common to use ingress filtering.
```

Reply:
https://github.com/vdaas/vald/pull/1783/commits/0c006116b16c5050a6204aa32b3d4680c9a153ce